test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -0,0 +1,235 @@
+        readerGroupManager.createReaderGroup(readerGroupId, readerGroupConfig);
+        @Cleanup
+        EventStreamReader<String> reader = clientFactory.createReader(readerGroupId + "1", readerGroupId,
+                new UTF8StringSerializer(), ReaderConfig.builder().build());
+        // Before sealing of the stream it must write to the existing stream.
+        writer.writeEvent("0").join();
+        assertTrue(streamManager.sealStream(scopeName, streamName));
+        // It must complete Exceptionally because stream is sealed already.
+        assertTrue(writer.writeEvent("1").completeExceptionally(new IllegalStateException()));
+        // But there should not be any problem while reading from a sealed stream.
+        assertEquals(reader.readNextEvent(READER_TIMEOUT_MS).getEvent(), "0");
+    }
+
+    /**
+     * This method attempts to create, delete and validate the existence of a scope.
+     * @throws DeleteScopeFailedException if unable to seal and delete a stream.
+     */
+    private void checkDeleteScope() throws DeleteScopeFailedException {
+        String scopeName = "scope-delete-test-scope";
+        String streamName = "scope-delete-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        assertTrue(streamManager.checkScopeExists(scopeName));
+
+        assertTrue(streamManager.deleteScopeRecursive(scopeName));
+
+        //validating whether the scope deletion is successful, expression must be false.
+        assertFalse(streamManager.checkScopeExists(scopeName));
+    }
+
+    /**
+     * This method is trying to create a stream and writing a couple of events.
+     * Then deleting the newly created stream and validating it.
+     */
+    private void checkStreamDelete() {
+        String scopeName = "stream-delete-test-scope";
+        String streamName = "stream-delete-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        assertTrue(streamManager.checkStreamExists(scopeName, streamName));
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(), EventWriterConfig.builder().build());
+        // Before sealing of the stream it must write to the existing stream.
+        writer.writeEvent("0").join();
+        assertTrue(streamManager.sealStream(scopeName, streamName));
+
+        assertTrue(streamManager.deleteStream(scopeName, streamName));
+        assertFalse(streamManager.checkStreamExists(scopeName, streamName));
+    }
+
+    public static void main(String[] args) throws DeleteScopeFailedException {
```

Comment:
If you want to use a parameterizable endpoint for Controller, you will need to do it here.